book/src/pong-tutorial/pong-tutorial-04.md

```diff
@@ -0,0 +1,365 @@
+# use amethyst::ecs::World;
+# use amethyst::core::transform::Transform;
+# use amethyst::core::cgmath::Vector3;
+# use amethyst::ecs::prelude::{Component, DenseVecStorage};
+# pub struct Ball {
+#    pub velocity: [f32; 2],
+#    pub radius: f32,
+# }
+# impl Component for Ball {
+#    type Storage = DenseVecStorage<Self>;
+# }
+# const PADDLE_HEIGHT: f32 = 16.0;
+# const PADDLE_WIDTH: f32 = 4.0;
+# const SPRITESHEET_SIZE: (f32, f32) = (8.0, 16.0);
+# const BALL_RADIUS: f32 = 2.0;
+# const BALL_VELOCITY_X: f32 = 75.0;
+# const BALL_VELOCITY_Y: f32 = 50.0;
+# const ARENA_HEIGHT: f32 = 100.0;
+# const ARENA_WIDTH: f32 = 100.0;
+/// Initialises one ball in the middle-ish of the arena.
+fn initialise_ball(world: &mut World, sprite_sheet_handle: SpriteSheetHandle) {
+    // Create the translation.
+    let mut local_transform = Transform::default();
+    local_transform.translation = Vector3::new(ARENA_WIDTH / 2.0, ARENA_HEIGHT / 2.0, 0.0);
+
+    // Assign the sprite for the ball
+    let sprite_render = SpriteRender {
+        sprite_sheet: sprite_sheet_handle,
+        sprite_number: 1, // ball is the second sprite on the sprite sheet
+        flip_horizontal: false,
+        flip_vertical: false,
+    };
+
+    world
+        .create_entity()
+        .with(sprite_render)
+        .with(Ball {
+            radius: BALL_RADIUS,
+            velocity: [BALL_VELOCITY_X, BALL_VELOCITY_Y],
+        })
+        .with(local_transform)
+        .build();
+}
+```
+
+In [a previous chapter][pong_02_drawing] we saw how to load a sprite sheet
+and get things drawn on the screen. Remember sprite sheet information
+is stored in `pong_spritesheet.ron`, and the ball sprite was the
+second one, whose index is `1`.
+
+Finally, let's make sure the code is working as intended by updating the `on_start` method:
+
+```rust,no_run,noplaypen
+# extern crate amethyst;
+# use amethyst::prelude::*;
+# use amethyst::renderer::{TextureHandle, SpriteSheetHandle};
+# use amethyst::ecs::World;
+# struct Paddle;
+# impl amethyst::ecs::Component for Paddle {
+#   type Storage = amethyst::ecs::VecStorage<Self>;
+# }
+# struct Ball;
+# impl amethyst::ecs::Component for Ball {
+#   type Storage = amethyst::ecs::VecStorage<Self>;
+# }
+# fn initialise_ball(world: &mut World, sprite_sheet_handle: SpriteSheetHandle) { }
+# fn initialise_paddles(world: &mut World, spritesheet: SpriteSheetHandle) { }
+# fn initialise_camera(world: &mut World) { }
+# fn load_sprite_sheet(world: &mut World) -> SpriteSheetHandle { unimplemented!() }
+# struct MyState;
+# impl<'a, 'b> SimpleState<'a, 'b> for MyState {
+fn on_start(&mut self, data: StateData<GameData>) {
+    let world = data.world;
+
+    // Load the spritesheet necessary to render the graphics.
+    let sprite_sheet_handle = load_sprite_sheet(world);
+
+    world.register::<Paddle>();
+    world.register::<Ball>(); // <- add this line
+
+    initialise_ball(world, sprite_sheet_handle.clone()); // <- and this line
```

Comment:
`and this line` should be `add this line`

Reply:
Does it sounds more natural even though `add this line` is already on line 127?